hera_pspec/pspecdata.py

```diff
@@ -83,12 +87,17 @@ def __init__(self, dsets=[], wgts=None, dsets_std=None, labels=None,
             dsets_std = None
 
         # Store the input UVData objects if specified
-        if len(dsets) > 0:
+        if dsets is not None and len(dsets) > 0:
```

Comment:
There is no way `dsets` can be `None` here since you have the condition `dsets = [] if dsets is None else dsets` above

Reply:
Good point!